modules/key-card/src/generateQrData.ts

```diff
@@ -1,19 +1,19 @@
-import { BaseCoin } from '@bitgo/statics';
+import { BaseCoin, KeyCurve } from '@bitgo/statics';
 import { Keychain } from '@bitgo/sdk-core';
 import { encrypt } from '@bitgo/sdk-api';
 import * as assert from 'assert';
 
 export interface GenerateQrDataParams {
   // The backup keychain as it is returned from the BitGo API upon creation
-  backupKeychain: Keychain;
+  backupKeychain?: Keychain;
   // The name of the 3rd party provider of the backup key if neither the user nor BitGo stores it
   backupKeyProvider?: string;
   // The key id of the backup key, only used for cold keys
   backupMasterKey?: string;
   // The BitGo keychain as it is returned from the BitGo API upon creation
   bitgoKeychain: Keychain;
   // The coin of the wallet that was/ is about to be created
-  coin: Readonly<BaseCoin>;
+  coin?: Readonly<BaseCoin>;
```

Comment:
Instead of marking fields as optional, can we instead make the params be a union of accepted types?:
```
type GenerateQrDataParams = GenerateQrDataForKeychainType1Params | GenerateQrDataForKeychainType2Params ...
```

Reply:
+1, this makes it easier to identify valid flows.